roles/validations/tasks/xmlcreate.yml

```diff
@@ -0,0 +1,58 @@
+---
+- name: Get validation start time from system
+  ansible.builtin.command:
+    cmd: date +%Y-%m-%d-%H:%M:%S.%3N
+  register: individual_validation_start_time
+
+- name: Run a validation
+  ansible.builtin.include_tasks: "{{ item }}"
+
+- name: Get validation end time from system
+  ansible.builtin.command:
+    cmd: date +%Y-%m-%d-%H:%M:%S.%3N
+  register: individual_validation_end_time
+
+- name: Calculate validation total run time
+  ansible.builtin.set_fact:
+    individual_validator_run_time: "{{((individual_validation_end_time.stdout | to_datetime('%Y-%m-%d-%H:%M:%S.%f')) - (individual_validation_start_time.stdout | to_datetime('%Y-%m-%d-%H:%M:%S.%f'))).total_seconds()}}"
+
+- name: Increment test count after validation execution
+  ansible.builtin.set_fact:
+    validations_executed: "{{ validations_executed | int + 1 }}"
+
+- name: Make new xml line list entry
+  ansible.builtin.set_fact:
+    new_xml_line_list_entry: ['  <testcase classname="validations.{{item | dirname}}" name="{{item | basename}}" time="{{ individual_validator_run_time }}"/>']
+  when: '"passed" in validator_status'
+
+- name: Add new line to xml line list
+  ansible.builtin.set_fact:
+    xml_line_list: "{{ xml_line_list + new_xml_line_list_entry }}"
+  when: '"passed" in validator_status'
+
+- name: Add testcase and failure reason to xml line list when a validation fails
+  when: '"failed" in validator_status'
+  block:
+    - name: Make new xml line list entry
+      ansible.builtin.set_fact:
+        new_xml_line_list_entry: ['  <testcase classname="validations.{{item | dirname}}" name="{{item | basename}}" time="{{ individual_validator_run_time }}">']
```

Comment:
Details like the indentation handling you do here is one of the reasons I added this comment:
https://github.com/openstack-k8s-operators/ci-framework/pull/2456/files#diff-745f651c7f88527b40991eca9fbe69b880b93734ee5a24b140a3fcac8becac2aR87

Reply:
Latest version uses junitparser to generate xml